vaadin-date-time-picker-flow-parent/vaadin-date-time-picker-flow/src/main/java/com/vaadin/flow/component/datetimepicker/DateTimePicker.java

```diff
@@ -374,7 +385,10 @@ public String getTimePlaceholder() {
      *            an hour evenly
      */
     public void setStep(Duration step) {
-        timePicker.setStep(step);
+        Objects.requireNonNull(step, "Step cannot be null");
+
+        getElement().setProperty("step",
+                StepsUtil.convertDurationToStepsValue(step));
```

Comment:
Adopted the same logic for setting the step from the TimePicker Flow component. The previous solution of setting the step on the time picker itself was flawed, as the date time picker web component has its own step property and related logic. This lead to a different bug: https://github.com/vaadin/flow-components/issues/2101

Setting the step on the date time picker component itself fixes the issue.